File: common/src/main/java/dev/obscuria/fragmentum/v2/core/common/signal/_Signal5.java
package dev.obscuria.fragmentum.v2.core.common.signal;

import dev.obscuria.fragmentum.v2.api.common.signal.Signal5;

public final class _Signal5<P1, P2, P3, P4, P5> extends _Signal<Signal5.Listener<P1, P2, P3, P4, P5>> implements Signal5<P1, P2, P3, P4, P5> {

    @Override public void emit(P1 p1, P2 p2, P3 p3, P4 p4, P5 p5) {
        emitInternal(it -> it.consume(p1, p2, p3, p4, p5));
    }
}

File: api/src/main/kotlin/dev/obscuria/fragmentum/tools/signal/Signal5.kt
@file:Suppress("unused", "UNUSED_PARAMETER", "RedundantNullableReturnType")

package dev.obscuria.fragmentum.tools.signal

class Signal5<P1, P2, P3, P4, P5> : Signal<Signal5.Listener<P1, P2, P3, P4, P5>>() {

    fun emit(p1: P1, p2: P2, p3: P3, p4: P4, p5: P5): Unit = error("API stub")

    fun interface Listener<P1, P2, P3, P4, P5> {

        fun consume(p1: P1, p2: P2, p3: P3, p4: P4, p5: P5)
    }
}